app/api/items/leave.feature

```diff
@@ -0,0 +1,149 @@
+      | 203             | 102            | 2019-12-31 23:59:59 |
+      | 203             | 111            | 9999-12-31 23:59:59 |
+    And the groups ancestors are computed
+    And the database has the following table 'items':
+      | id | url                                                                     | type    | allows_multiple_attempts | participants_group_id | default_language_tag |
+      | 10 | null                                                                    | Chapter | 0                        | 201                   | fr                   |
+      | 50 | http://taskplatform.mblockelet.info/task.html?taskId=403449543672183936 | Task    | 0                        | 202                   | fr                   |
+      | 60 | http://taskplatform.mblockelet.info/task.html?taskId=403449543672183936 | Course  | 1                        | 203                   | fr                   |
+    And the database has the following table 'items_items':
+      | parent_item_id | child_item_id | child_order |
+      | 10             | 60            | 1           |
+    And the database has the following table 'items_ancestors':
+      | ancestor_item_id | child_item_id |
+      | 10               | 60            |
+    And the database has the following table 'permissions_generated':
+      | group_id | item_id | can_view_generated       |
+      | 101      | 50      | content                  |
+      | 102      | 60      | content                  |
+      | 111      | 10      | content_with_descendants |
+      | 111      | 50      | content_with_descendants |
+    And the database has the following table 'attempts':
+      | id | participant_id | root_item_id | parent_attempt_id | allows_submissions_until | ended_at            |
+      | 0  | 101            | 10           | null              | 9999-12-31 23:59:59      | null                |
+      | 0  | 102            | 10           | null              | 9999-12-31 23:59:59      | null                |
+      | 0  | 111            | 10           | null              | 9999-12-31 23:59:59      | null                |
+      | 1  | 101            | 50           | null              | 9999-12-31 23:59:59      | null                |
+      | 1  | 102            | 50           | null              | 9999-12-31 23:59:59      | null                |
+      | 1  | 111            | 50           | null              | 9999-12-31 23:59:59      | null                |
+      | 2  | 101            | 60           | 0                 | 9999-12-31 23:59:59      | null                |
+      | 2  | 102            | 60           | 0                 | 2019-12-31 23:59:59      | 2019-12-31 23:59:59 |
+      | 2  | 111            | 60           | 0                 | 9999-12-31 23:59:59      | null                |
+
+  Scenario: User is able to end an attempt for his self group
+    Given I am the user with id "111"
+    When I send a DELETE request to "/attempts/0"
```

Comment:
Ah it is a good example of something we should probably not allow. It should not be allowed to end the default attempt (id 0), otherwise the user is completely stuck. Anyway it does not really make sense to end attempt 0 which was implicit.

Reply:
fixed